### PR TITLE
Modernise for newer target frameworks

### PR DESCRIPTION
**Pull request by jetersen**
- Target newer frameworks including net5 and netstandard2.1 for greater compatibility
- Include symbols package
- Use License Expression
- Add Package References for reference assemblies for NET Framework
- Add SourceLink
- Add GitVersion Properties
- Use GitHub Actions
- Remove legacy global.json
- Remove .travis.yml
- Remove Dockerfile
- Remove Makefile as it was only used for docker tests

Hi @omar 
Would you prefer to use license file over license expression?

I created GitHub action that will push to nuget.org when you publish a GitHub Release, as long as you add an NuGet API key to the [GitHub repo secrets](https://github.com/omar/ByteSize/settings/secrets/actions) with the `NUGET_API_KEY` name.
The action uses GitVersion to retrieve version based on git tags, so no need to make changes to csproj file to deploy a new version, simply create a GitHub Release and the rest will be taken care of.

I have used the same GitHub action for https://github.com/nmklotas/GitLabApiClient and https://github.com/Specshell/specshell.software.ndde

Reason to target net standard 2.0 and 2.1 can be found here: https://docs.microsoft.com/en-us/dotnet/standard/frameworks#latest-versions

### CODE REVIEW

**Comment by jetersen** [comment]
Seems the project does not allow GitHub Actions to run.
[You can enable them here](https://github.com/omar/ByteSize/settings/actions)

I have the build running here: https://github.com/jetersen/ByteSize/runs/1420624006

**Comment by omar** [comment]
@jetersen thank you for putting this PR together. This all looks great. One question I had was about the supported framework. As I understand it, targeting netstandard1.0 would give us the widest support of customers. If we move things to a minimum of  netstandard2.0, we would lose support for previous frameworks. That's something I'd like to avoid given there's no immediate need to require netstandard 2.0

**Comment by jetersen** [comment]
@omar sounds reasonable 👍

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1'
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '5.0'
```
Comment (jetersen) [inline .github/workflows/ci.yml:38]: FYI these lines can be removed once this lands: https://github.com/actions/virtual-environments/issues/1891

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name: DotNet
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
```
Comment (jetersen) [inline .github/workflows/ci.yml:9]: Use default branch macro.

```suggestion
      - $default-branch
  push:
    branches:
      - $default-branch
```

**Comment by jetersen** [comment]
@omar you'd want to enable the GitHub actions here: https://github.com/omar/ByteSize/settings/actions

**Comment by omar** [comment]
I have that enabled. Still not seeing anything running.
image

**Pull request merged**